### PR TITLE
[wip] typed param example

### PR DESCRIPTION
**Pull request by mickjermsurawong-stripe**
- This is illustrative PR to explain https://github.com/stripe/stripe-java/pull/679
- Please see the tests using the typed params in commit with message prefixed with "test:..."

### CODE REVIEW

File: src/test/java/com/stripe/functional/SkuTest.java [hunk 1]
```diff
@@ -26,6 +28,45 @@ private Sku getSkuFixture() throws StripeException {
...
+    verifyRequest(
+        ApiResource.RequestMethod.POST,
+        "/v1/skus",
+        params
+    );
+  }
+
+  @Test
+  public void testCreateWithParams() throws StripeException {
+    SkuCreateParams.Inventory inventory = SkuCreateParams.Inventory.builder()
```
Comment (mickjermsurawong-stripe) [inline src/test/java/com/stripe/functional/SkuTest.java]: Inner inventory object passed as argument to the parent param builder.

File: src/test/java/com/stripe/functional/InvoiceTest.java [hunk 2]
```diff
@@ -69,6 +76,71 @@ public void testUpdate() throws StripeException {
...
+
+    Map<String, Object> paramsWithEmpty = new HashMap<>();
+    paramsWithEmpty.put("custom_fields", null);
+    paramsWithEmpty.put("tax_percent", null);
+
+    assertNotNull(updatedInvoice);
+    verifyRequest(
+        ApiResource.RequestMethod.POST,
+        String.format("/v1/invoices/%s", invoice.getId()),
+        paramsWithEmpty
```
Comment (mickjermsurawong-stripe) [inline src/test/java/com/stripe/functional/InvoiceTest.java]: Verify that "EMPTY" enums are serialized as nulls

File: src/test/java/com/stripe/functional/PaymentIntentTest.java [hunk 3]
```diff
@@ -55,6 +64,59 @@ public void testRetrieve() throws StripeException {
...
+    // expansion fields and the resulting verification is copied
+    // from `model/PaymentIntentTest`
+    PaymentIntentRetrieveParams typedParams = PaymentIntentRetrieveParams.builder()
+        .addExpand("application")
+        .addExpand("customer")
+        .addExpand("on_behalf_of")
+        .addExpand("review")
+        .addExpand("source")
+        .addExpand("transfer_data.destination")
+        .build();
```
Comment (mickjermsurawong-stripe) [inline src/test/java/com/stripe/functional/PaymentIntentTest.java]: Basic expand params

**Comment by brandur-stripe** [comment]
OMG this is freaking awesome. Everyone using stripe-java is going to love you for this, haha.

**Comment by mickjermsurawong-stripe** [comment]
thanks @brandur-stripe! :)

**Pull request closed without merging**